### ISSUE TITLE
double quotes not escaped in JSON string values

### ISSUE THREAD

**Issue by grkblood13** [mkrufky/libdvbtee#50]
double quotes not escaped in JSON string values causing invalid JSON output

**Comment by mkrufky** [comment]
I'd love to fix this if you can show me an example of the issue.   Otherwise, patches are welcome :-)

**Comment by grkblood13** [comment]
Sure, check out this pastebin. Line 62 shows the EPG data breaking in the JSON output. The JSON is at the bottom. https://pastebin.com/jXQS26A3

**Comment by mkrufky** [comment]
I believe that this will work properly if you use the node.js module, [node-dvbtee](https://github.com/mkrufky/node-dvbtee) ...  I don't recall why it was fixed there but not in the CLI app, but if using the node.js module is an option for you, it may be a better route.

**Comment by grkblood13** [comment]
Any chance this will be fixed in the cli version? I would contribute but I'm all that experienced with C++ programming.

**Comment by grkblood13** [comment]
Is there any way to associate the EIT and ETT tables printed from the node-dvbtee packaged you mentioned with the virtual channel it's meant for? I can't seem to find any field to make the correlation.

**Comment by mkrufky** [comment]
> Any chance this will be fixed in the cli version? I would contribute but I'm not all that experienced with C++ programming.

Not likely to happen immediately.  It needs to be done in such a way that preserves the original behavior by default -- others use the software to extract binary blobs from the streams as-is, and any escaping must be optional.

This was done in `node-dvbtee` and it works properly there.  If you'd like to see this logic brought the the CLI, please confirm that the way it's done in `node-dvbtee` works for you.

**Comment by mkrufky** [comment]
> Is there any way to associate the EIT and ETT tables printed from the node-dvbtee packaged you mentioned with the virtual channel it's meant for? I can't seem to find any field to make the correlation.

When using `node-dvbtee`, you will have to associate the data from the related tables to each other yourself.  I had planned on adding such utility natively to `node-dvbtee` but hadn't gotten around to it yet.

**Comment by grkblood13** [comment]
What's the significance of the etmid thats in the ett tables? I've figured out how to pair the eit tables with the proper data but not how to match up the ett tables with their corresponding eit programme event. Im thinking maybe I can use this "etmId" somehow? My main problem is that I can't find anywhere in the parsed data where these etmIds are referenced anywhere.

**Comment by grkblood13** [comment]
Nevermind, I figured it out after some decimal to binary conversions. I'm going to leave this open though since the original issue remains unresolved. Thanks for the suggestions!

**Comment by bkoehm** [comment]
I opened pull request #60 to escape double quotes and backslashes in JSON values.  I tested this with a local ATSC .ts file using the -F, -j and -e options.  I wanted to test the web server output, but I was unable to figure out how to get the web server to output as JSON (only as HTML).  If someone tells me how to do that, I can test the web server output too.  

Also I want to note that I found at least one other place in `object.cpp` `Object::toJson()` that seems to be outputting JSON values.  I do not know what calls this or what JSON this produces, so I left it alone.  But if it turns out this needs to be escaped too, the same function in my pull request can be used to escape it.
